src/widgets/beszel/proxy.js

```diff
@@ -72,8 +72,15 @@ export default async function beszelProxyHandler(req, res) {
         },
       });
 
-      if ([400, 403].includes(status)) {
-        logger.debug(`HTTP ${status} retrieving data from Beszel, logging in and trying again.`);
+      const badRequest = [400, 403].includes(status);
+      const isEmpty = data.includes("[]");
```

Comment:
```suggestion
      const text = data.toString("utf-8");
      let isEmpty = false;

      try {
        const json = JSON.parse(text);
        isEmpty = Array.isArray(json.items) && json.items.length === 0;
      } catch (err) {
        logger.debug("Failed to parse Beszel response JSON:", err);
      }
```

Maybe seems overkill but more precise, less brittle I think

Reply:
Perfect, thanks :+1: